File: 0621-task-scheduler/0621-task-scheduler.java
class Solution {
    public int leastInterval(char[] tasks, int n) {
        int[] freq = new int[26];
        for (char task : tasks) {
            freq[task - 'A']++;
        }

        int maxCount = 0;
        for (int f : freq) {
            maxCount = Math.max(maxCount, f);
        }

        int maxCountTask = 0;
        for (int f : freq) {
            if (f == maxCount) {
                maxCountTask++;
            }
        }

        int frameTime = (maxCount - 1) * (n + 1) + maxCountTask;

        return Math.max(frameTime, tasks.length);
    }
}

File: 0621-task-scheduler/0621-task-scheduler.kt
class Solution {
    fun leastInterval(tasks: CharArray, n: Int): Int {
        val frequency = IntArray(26)
        for (task in tasks) {
            frequency[task - 'A']++
        }

        val maxFrequency = frequency.max()
        val tasksWithMaxFrequency = frequency.count { it == maxFrequency }
        val frameTime = (maxFrequency - 1) * (n + 1) + tasksWithMaxFrequency

        return maxOf(frameTime, tasks.size)
    }
}